packages/signaldb/src/SyncManager/applyChanges.ts

```diff
@@ -16,9 +17,19 @@ export default function applyChanges<ItemType extends BaseItem<IdType>, IdType>(
   items.forEach(item => collection.insert(item))
   changes.forEach((change) => {
     if (change.type === 'insert') {
-      collection.insert(change.data)
+      const itemExists = collection.findOne({ id: change.data.id } as Record<string, any>)
+      if (itemExists) {
+        collection.updateOne({ id: change.data.id } as Record<string, any>, { $set: change.data })
+      } else {
+        collection.insert(change.data)
+      }
     } else if (change.type === 'update') {
-      collection.updateOne({ id: change.data.id } as Record<string, any>, change.data.modifier)
+      const itemExists = collection.findOne({ id: change.data.id } as Record<string, any>)
+      if (itemExists) {
+        collection.updateOne({ id: change.data.id } as Record<string, any>, change.data.modifier)
```

Comment:
it took me a moment to realize that the difference between the insert and the update is the `data.modifier` vs `data`. I am also curious about the cast to `Record<string, any>` given that the `collection` and `changes` should be based on the same Generic type provided during the collection creation.

Reply:
I’m also not really happy with this code. It was a bit late yesterday 😀
The only reason for the cast to `Record<string, any>` is to please the TypeScript Compiler

Reply:
I refactored this code to make it more readable